Library/Homebrew/utils/curl.rb

```diff
@@ -494,9 +498,14 @@ def curl_http_content_headers_and_checksum(
       T.must(file).unlink
     end
 
+    def curl_version
```

Comment:
We use `HOMEBREW_MINIMUM_CURL_VERSION` and run `curl -V` in `brew.sh` always on Linux. Might be nicer to use `curl_version_output` to unconditionally set `HOMEBREW_CURL_VERSION` and use that here rather than shelling out here.

Reply:
I moved the existing logic from `curl_supports_fail_with_body?` and it looked like the code was set up to be re-run on different `curl`s rather than just the main `curl`. Haven't found how this is used yet.

I think minimal impact change can be merged first to fix open issue and then look if a refactor is possible.

Reply:
[Added here](https://github.com/Homebrew/brew/pull/17356) so that older macOS versions would use brewed curl for livechecks involving page content.

Reply:
@cho-m Cool, fine with me to merge as-is then!